### PR TITLE
doc: Added links to folders and headers mentioned

### LINKED ISSUE #10110: Adding links in `CONTRIBUTING.md` to sub folders and headers mentioned
This small change would make it easier for viewers of the file to find folders and headers mentioned. I will reference a PR to this soon.

### PR DESCRIPTION
**Pull request by UnrealApex**
## Summary of the Pull Request
Added links to folders and headers mentioned.
 
## References
N/A

## PR Checklist
* [x] Closes #10110 
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [x] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
Added links inside `CONTRIBUTING.md` leading users to headers and sub folders mentioned.

## Validation Steps Performed
Tested verifying links working.

### CODE REVIEW

**Review by zadjii-msft — APPROVED** [review]
sure, thanks!

**Comment by UnrealApex** [comment]
I accidently used the wrong issue template, my bad.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -111,13 +111,13 @@ However, some issues/features will require careful thought & formal design befor
 
 Specs help collaborators discuss different approaches to solve a problem, describe how the feature will behave, how the feature will impact the user, what happens if something goes wrong, etc. Driving towards agreement in a spec, before any code is written, often results in simpler code, and less wasted effort in the long run.
 
-Specs will be managed in a very similar manner as code contributions so please follow the "Fork, Branch and Create your PR" below.
+Specs will be managed in a very similar manner as code contributions so please follow the "[Fork, Branch and Create your PR" below](CONTRIBUTING.md#fork-clone-branch-and-create-your-pr)".
```
Comment (DHowett) [inline CONTRIBUTING.md:114]: Minor rewording -- we apparently messed up the grammar here!

```suggestion
Specs will be managed in a very similar manner as code contributions so please follow the "[Fork, Branch and Create your PR](CONTRIBUTING.md#fork-clone-branch-and-create-your-pr)" section below.
```

**Review by DHowett — APPROVED** [review]
Thanks for doing this! I really appreciate the cleanup :)

**Pull request merged**